server/src/main/java/io/crate/execution/engine/indexing/ShardDMLExecutor.java

```diff
@@ -134,19 +141,40 @@ private String resolveNodeId(TReq request) {
     private CompletableFuture<TAcc> executeBatch(TReq request) {
         ConcurrencyLimit nodeLimit = nodeLimits.get(resolveNodeId(request));
         long startTime = nodeLimit.startSample();
-        FutureActionListener<ShardResponse, TAcc> listener = new FutureActionListener<>((a) -> {
-            nodeLimit.onSample(startTime, false);
-            TAcc acc = collector.supplier().get();
-            collector.accumulator().accept(acc, a);
-            return acc;
-        });
+        CompletableFuture<TAcc> future = new CompletableFuture<>();
+        var listener = new ActionListener<ShardResponse>() {
+
+            @Override
+            public void onResponse(ShardResponse response) {
+                nodeLimit.onSample(startTime, false);
```

Comment:
consider extracting method? but also fine since it's only a few lines repeated in `onFailure()`

Reply:
Not worth it